live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/PreSignedUrlRequesterTest.java

```diff
@@ -0,0 +1,202 @@
+
+@ExtendWith(MockitoExtension.class)
+class PreSignedUrlRequesterTest
+{
+    private static final int STATUS_CODE_201 = 201;
+    private static final String STORAGE_LOCATION = "http://dummy-url";
+    private static final String RESPONSE_BODY = "{\"" + STORAGE_LOCATION_PROPERTY + "\": \"" + STORAGE_LOCATION + "\"}";
+
+    @Mock
+    CamelContext camelContextMock;
+    @Mock
+    ProducerTemplate producerTemplateMock;
+    @Mock
+    Exchange exchangeMock;
+    @Mock(strictness = Mock.Strictness.LENIENT)
+    Message messageMock;
+    @Mock
+    ObjectMapper objectMapperMock;
+
+    @InjectMocks
+    PreSignedUrlRequester preSignedUrlRequester;
+
+    @BeforeEach
+    void setUp()
+    {
+        given(camelContextMock.createProducerTemplate()).willReturn(producerTemplateMock);
+        given(producerTemplateMock.send(any(String.class), any(Processor.class))).willReturn(exchangeMock);
+        given(exchangeMock.getMessage()).willReturn(messageMock);
+        given(messageMock.getHeader(any(String.class), any(Class.class))).willReturn(STATUS_CODE_201);
+        given(messageMock.getBody(any())).willReturn(RESPONSE_BODY);
+    }
+
+    @Test
+    void testRequestStorageLocation() throws IOException
+    {
+        // given
+        StorageLocationRequest request = createStorageLocationRequestMock();
+        given(objectMapperMock.readValue(any(String.class), any(TypeReference.class))).willReturn(Map.of(STORAGE_LOCATION_PROPERTY, STORAGE_LOCATION));
+
+        // when
+        URL url = preSignedUrlRequester.requestStorageLocation(request);
+
+        // then
+        then(messageMock).should().getHeader(Exchange.HTTP_RESPONSE_CODE, Integer.class);
+        then(messageMock).should().getBody(String.class);
+        then(objectMapperMock).should().readValue(eq(RESPONSE_BODY), anyTypeReferenceOfMap());
+        assertThat(url).asString().isEqualTo(STORAGE_LOCATION);
+    }
+
+    @Test
+    void testRequestStorageLocation_invalidResponseStatusCode()
+    {
+        // given
+        StorageLocationRequest request = createStorageLocationRequestMock();
+        given(messageMock.getHeader(any(String.class), any(Class.class))).willReturn(500);
+
+        // when
+        Throwable thrown = catchThrowable(() -> preSignedUrlRequester.requestStorageLocation(request));
+
+        // then
+        then(objectMapperMock).shouldHaveNoInteractions();
+        assertThat(thrown)
+                .isInstanceOf(LiveIngesterRuntimeException.class)
+                .hasMessageContaining("received:", 500);
+    }
+
+    @Test
+    void testRequestStorageLocation_missingStorageLocationPropertyInResponse() throws IOException
+    {
+        // given
+        StorageLocationRequest request = createStorageLocationRequestMock();
+        String unexpectedProperty = "unexpectedProperty";
+        String responseBodyWithoutUrl = "{\"" + unexpectedProperty + "\": \"" + STORAGE_LOCATION + "\"}";
```

Comment:
I would propose to use multiline string as it will be much more readable:
```
String invalidResponseBody = String.format("""
                {"%s": "%s"}
                """, "unexpectedProperty", STORAGE_LOCATION);
```

Reply:
Imo this example is a bit of overkill, but I've fixed a bit response body creation.